File: examples/typescript/02_wallet.ts
// Scenarios 4-7: create/load a descriptor wallet, generate an address,
// check balance, and list unspent outputs. Regtest recommended.
import { KingPepeClient, formatKpepe, rpcAmountToBaseUnits } from "@kingpepe/sdk";

const WALLET = "example-shop";

async function main() {
  const client = KingPepeClient.fromEnv();

  // Create the wallet if it doesn't exist, otherwise load it.
  const wallets = await client.listWallets();
  if (!wallets.includes(WALLET)) {
    try {
      await client.createWallet(WALLET, { descriptors: true });
    } catch {
      await client.loadWallet(WALLET);
    }
  }
  const wallet = client.withWallet(WALLET);

  const address = await wallet.getNewAddress("example-label");
  console.log("New receiving address:", address);

  const balances = await wallet.getBalances();
  const trusted = rpcAmountToBaseUnits(balances.mine.trusted);
  console.log("Trusted balance:", formatKpepe(trusted), "KPEPE");

  const unspent = await wallet.listUnspent(0);
  console.log(`Unspent outputs: ${unspent.length}`);
  for (const u of unspent.slice(0, 5)) {
    console.log(`  ${u.txid}:${u.vout}  ${formatKpepe(rpcAmountToBaseUnits(u.amount))} KPEPE  conf=${u.confirmations}`);
  }
}

main().catch((err) => {
  console.error("error:", err.message);
  process.exit(1);
});

File: examples/python/02_wallet.py
"""Scenarios 4-7: create/load a descriptor wallet, generate an address,
check balance, and list unspent outputs. Regtest recommended."""

from __future__ import annotations

from kingpepe_sdk import KingPepeClient, format_kpepe, rpc_amount_to_base_units

WALLET = "example-shop"


def main() -> None:
    client = KingPepeClient.from_env()

    if WALLET not in client.list_wallets():
        try:
            client.create_wallet(WALLET, descriptors=True)
        except Exception:
            client.load_wallet(WALLET)
    wallet = client.with_wallet(WALLET)

    address = wallet.get_new_address("example-label")
    print("New receiving address:", address)

    balances = wallet.get_balances()
    trusted = rpc_amount_to_base_units(balances["mine"]["trusted"])
    print("Trusted balance:", format_kpepe(trusted), "KPEPE")

    unspent = wallet.list_unspent(0)
    print(f"Unspent outputs: {len(unspent)}")
    for u in unspent[:5]:
        amt = format_kpepe(rpc_amount_to_base_units(u["amount"]))
        print(f"  {u['txid']}:{u['vout']}  {amt} KPEPE  conf={u['confirmations']}")


if __name__ == "__main__":
    main()
